contracts/contracts/harvest/BaseHarvester.sol

```diff
@@ -328,31 +428,255 @@ abstract contract BaseHarvester is Governable {
     }
 
     /**
-     * @dev Swap all supported swap tokens for stablecoins via Uniswap. And send the incentive part
-     * of the rewards to _rewardTo address.
-     * @param _rewardTo Address where to send a share of harvest rewards to as an incentive
-     *      for executing this function
+     * @dev Swap a reward token for the base token on the configured
+     *      swap platform. The token must have a registered price feed
+     *      with the price provider
+     * @param _swapToken Address of the token to swap
+     * @param _rewardTo Address where to send the share of harvest rewards to
      */
-    function _swap(address _rewardTo) internal {
-        address[] memory allStrategies = IVault(vaultAddress)
-            .getAllStrategies();
-
-        for (uint256 i = 0; i < allStrategies.length; i++) {
-            IStrategy strategy = IStrategy(allStrategies[i]);
-            address[] memory rewardTokenAddresses = strategy
-                .getRewardTokenAddresses();
-
-            for (uint256 j = 0; j < rewardTokenAddresses.length; j++) {
-                _swap(rewardTokenAddresses[j], _rewardTo);
-            }
+    function _swap(address _swapToken, address _rewardTo) internal virtual {
+        RewardTokenConfig memory tokenConfig = rewardTokenConfigs[_swapToken];
+
+        /* This will trigger a return when reward token configuration has not yet been set
+         * or we have temporarily disabled swapping of specific reward token via setting
+         * doSwapRewardToken to false.
+         */
+        if (!tokenConfig.doSwapRewardToken) {
+            return;
+        }
+
+        address priceProvider = IVault(vaultAddress).priceProvider();
```

Comment:
idea: we could query for the priceProvider in a higher up function e.g. `_harvestAndSwap` and pass it down, so it is not fetched multiple times (for some gas savings). It will dirty up the code to some extent though

Reply:
It's only fetched once though

`harvestAndSwap` can no longer be called for multiple strategies. And `_harvest` doesn't use priceProvider. Only `_swap` does. 

The only other place we fetch and use the priceProvider is under `setRewardTokenConfig` method

Reply:
Ah, I get it now. Will make that change